src/core/public/notifications/toasts/global_toast_list.tsx

```diff
@@ -17,47 +17,46 @@
  * under the License.
  */
 
-import React, {
-  Component,
-} from 'react';
-import PropTypes from 'prop-types';
-
-import {
-  EuiGlobalToastList,
-  EuiPortal,
-} from '@elastic/eui';
-
-export class GlobalToastList extends Component {
-  constructor(props) {
-    super(props);
-
-    if (this.props.subscribe) {
-      this.props.subscribe(() => this.forceUpdate());
-    }
-  }
+import { EuiGlobalToastList, Toast } from '@elastic/eui';
+
+import React from 'react';
+import * as Rx from 'rxjs';
+
+interface Props {
+  toasts$: Rx.Observable<Toast[]>;
+  dismissToast: (t: Toast) => void;
+}
+
+interface State {
+  toasts: Toast[];
+}
 
-  static propTypes = {
-    subscribe: PropTypes.func,
-    toasts: PropTypes.array,
-    dismissToast: PropTypes.func.isRequired,
-    toastLifeTimeMs: PropTypes.number.isRequired,
+export class GlobalToastList extends React.Component<Props, State> {
+  public state: State = {
+    toasts: [],
   };
 
-  render() {
-    const {
-      toasts,
-      dismissToast,
-      toastLifeTimeMs,
-    } = this.props;
+  private subscription?: Rx.Subscription;
+
+  public componentDidMount() {
+    this.subscription = this.props.toasts$.subscribe(toasts => {
+      this.setState({ toasts });
+    });
+  }
+
+  public componentWillUnmount() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
 
+  public render() {
     return (
-      <EuiPortal>
-        <EuiGlobalToastList
-          toasts={toasts}
-          dismissToast={dismissToast}
-          toastLifeTimeMs={toastLifeTimeMs}
-        />
-      </EuiPortal>
+      <EuiGlobalToastList
+        toasts={this.state.toasts}
+        dismissToast={this.props.dismissToast}
+        toastLifeTimeMs={6000}
```

Comment:
Technically, the original implementation is [hard-coded](https://github.com/elastic/kibana/blob/master/src/ui/public/chrome/directives/kbn_chrome.js#L107) too, it's just not as obvious. I chose to hard-code this value to prevent toasts from behaving differently throughout Kibana, and to prevent people from loading them with too much content (allowing longer lifetimes allows for longer content).

Reply:
Hmm, didn't realize that APM had its own implementation of the global notifications list... I assume I can remove those and they can use the notifications from the new platform...

@sqren now that the toast notifications are available from the new platform you won't have to use angular to use them, and I won't have to worry that your global toast list will conflict with the new platform's global toast list, would you mind if I migrated APM to use that rather than its own instance?

Reply:
@spalger APM is using the EUI component. What are the advantages to having this in the platform instead using EUI directly?

Reply:
Oh, are you saying that there might be multiple `EuiGlobalToastList` on the page, because the platform also adds one?

Reply:
Exactly, the new platform will now be adding the `<EuiGlobalToastList/>` to every page, not just the Angular ones.

Reply:
Since APM isn't using our notification APIs, and there isn't anything that will cause double notifications today, I'm going to go ahead with this as-is and work on https://github.com/elastic/kibana/issues/21948 after this is merged.